docs/metrics.md

```diff
@@ -0,0 +1,561 @@
+
+* (T0, T1]
+  * attributes: {name = `apple`, color = `red`}, count: `1`
+  * attributes: {verb = `lemon`, color = `yellow`}, count: `2`
+* (T0, T2]
+  * attributes: {name = `apple`, color = `red`}, count: `1`
+  * attributes: {verb = `lemon`, color = `yellow`}, count: `2`
+* (T0, T3]
+  * attributes: {name = `apple`, color = `red`}, count: `6`
+  * attributes: {name = `apple`, color = `green`}, count: `2`
+  * attributes: {verb = `lemon`, color = `yellow`}, count: `12`
+
+Note that the start time is not advanced, and the exported values are the
+cumulative total of what happened since the beginning.
+
+### Example - Delta Aggregation Temporality
+
+If we aggregate and export the metrics using [Delta Aggregation
+Temporality](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#temporality):
+
+* (T0, T1]
+  * attributes: {name = `apple`, color = `red`}, count: `1`
+  * attributes: {verb = `lemon`, color = `yellow`}, count: `2`
+* (T1, T2]
+  * nothing since we do not have any measurement received
+* (T2, T3]
+  * attributes: {name = `apple`, color = `red`}, count: `5`
+  * attributes: {name = `apple`, color = `green`}, count: `2`
+  * attributes: {verb = `lemon`, color = `yellow`}, count: `10`
+
+Note that the start time is advanced after each export, and only the delta since
+last export is exported, allowing SDK to "forget" previous state.
+
+### Pre-Aggregation
+
+Using the [fruit example](#example), there are six measurements reported during
+the time range `(T2, T3]`. Instead of exporting each individual measurement
+event, the SDK aggregates them and exports only the summarized results. This
+process, illustrated in the following diagram, is known as pre-aggregation:
+
+```mermaid
+graph LR
+
+subgraph SDK
+    Instrument --> | Measurements | Pre-Aggregation[Pre-Aggregation]
+end
+
+subgraph Collector
+    Aggregation
+end
+
+Pre-Aggregation --> | Metrics | Aggregation
+```
+
+In addition to the in-process aggregation performed by the OpenTelemetry Rust
+Metrics SDK, further aggregations can be carried out by the Collector and/or the
+metrics backend.
+
+### Pre-Aggregation Benefits
+
+Pre-aggregation offers several advantages:
+
+1. **Reduced Data Volume**: Summarizes measurements before export, minimizing
+   network overhead and improving efficiency.
+2. **Predictable Resource Usage**: Ensures consistent resource consumption by
+applying [cardinality limits](#cardinality-limits) and [memory
+preallocation](#memory-preallocation) during SDK initialization. In other words,
+metrics storage/network needs remains fixed, irrespective of growing volume or
+changing traffic patterns.
+3. **Improved Performance**: Reduces computational load on downstream systems,
+   enabling them to focus on analysis and storage.
+
+> [!NOTE] There is no ability to export raw measurement events instead of using
```

Comment:
Maybe just use "measurements" instead of "measurement events"?
```suggestion
> [!NOTE] There is no ability to export raw measurements instead of using
```

Reply:
> instead of using pre-aggregation.

"You cannot opt out of pre-aggregation."